packages/framework/ai-collab/src/explicit-strategy/agentEditReducer.ts

```diff
@@ -124,18 +148,30 @@ export function applyAgentEdit(
 				if (allowedType.identifier === schemaIdentifier && typeof allowedType === "function") {
 					const simpleNodeSchema = allowedType as unknown as new (dummy: unknown) => TreeNode;
 					const insertNode = new simpleNodeSchema(treeEdit.content);
-					validator?.(insertNode);
+					try {
+						validator?.(insertNode);
+					} catch (error) {
+						if (error instanceof Error) {
+							throw new UsageError(error.message);
+						}
+						throw new Error("The node provided for insertion is not valid");
```

Comment:
Here we're completely swallowing the actual error and we'll never know what actually happened. Just `throw error` seems fine, to let the original error propagate.